hydra-node/test/Hydra/Chain/Direct/StateSpec.hs

```diff
@@ -144,6 +164,49 @@ spec = parallel $ do
     -- TODO: look into why this is failing
     propIsValid maxTxExecutionUnits (expectFailure . forAllFanout)
 
+  describe "ChainSyncHandler" $ do
+    prop "yields observed transactions rolling forward" $ do
+      forAllSt $ \(SomeOnChainHeadState -> st) tx -> do
+        let callback = \case
+              Rollback{} ->
+                fail "rolled back but expected roll forward."
+              Observation onChainTx ->
+                fst <$> observeSomeTx tx st `shouldBe` Just onChainTx
+        forAllBlind (genBlockAt 1 [tx]) $ \blk -> monadicIO $ do
+          headState <- run $ newTVarIO st
+          handler <- run $ newChainSyncHandler nullTracer callback headState
+          run $ onRollForward handler blk
+
+    prop "rollback rewind the chain state" $
+      forAllBlind genSequenceOfObservableBlocks $ \(st, blks) ->
+        forAll (choose (0, length blks - 1)) $ \ix -> do
+          let rollbackPoint = blockPoint (blks !! ix)
+          let callback = \case
+                Observation{} -> do
+                  pure ()
+                Rollback n -> n `shouldBe` fromIntegral ix
+          monadicIO $ do
+            headState <- run $ newTVarIO st
+            handler <- run $ newChainSyncHandler nullTracer callback headState
+            run $ mapM_ (onRollForward handler) blks
+            st' <- run $ readTVarIO headState
+            monitor $ counterexample $ "On-chain head state: " <> show st'
+            run $ onRollBackward handler rollbackPoint
```

Comment:
I fail to see how this test is checking that "rollback rewind the chain state"? Maybe the naming is wrong and what we test is that the `callback` is calling with `Rollback` when the chain sync does a `onRollBackward`?